GithubProfileWiki/Networking/ResponseHandler+Extension.swift

```diff
@@ -0,0 +1,34 @@
+//
+//  ResponseHandler+Extension.swift
+//  GithubProfileWiki
+//
+//  Created by ilter on 23.01.2022.
+//
+
+import Foundation
+
+struct ServiceError: Error, Codable {
+    let httpStatus: Int
+    let message: String
+}
+
+
+extension ResponseHandlerProtocol {
+    func defaultParseResponse<T: Codable>(data: Data, response: HTTPURLResponse) throws -> T {
+        let jsonDecoder = JSONDecoder()
+        jsonDecoder.keyDecodingStrategy = .convertFromSnakeCase
+        do {
+            let body = try jsonDecoder.decode(T.self, from: data)
+            
+            if response.statusCode == 200 {
+                return body
```

Comment:
i am not sure is this the correct way to handle errors? what if api returns an error with code 200

Reply:
You are right I will improve error handling cases 

- [ ] improve error handling on ResponseHandler

Reply:
But how can I add extra error conditions and handlers to this part? 
Also, did you mean like POST request's response code should be 201 and I can not return the body because of magic number 200 check